Formula/cabal-cache.rb

```diff
@@ -0,0 +1,20 @@
+class CabalCache < Formula
+  desc "Haskell cabal store caching tool"
+  homepage "https://github.com/haskell-works/cabal-cache"
+  url "https://hackage.haskell.org/package/cabal-cache-1.0.5.1/cabal-cache-1.0.5.1.tar.gz"
+  sha256 "4f464688f2d327d4f71c09697a95b3cb335dd29d0e173bb27675656906b1510a"
+  license "BSD-3-Clause"
+  head "git@github.com:haskell-works/cabal-cache.git", branch: "main"
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc@8.10" => :build
+
+  def install
+    system "cabal", "v2-update"
+    system "cabal", "v2-install", *std_cabal_v2_args
+  end
+
+  test do
+    assert_match(/^cabal-cache .*$/, shell_output("#{bin}/cabal-cache version"))
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The application depends on `cabal` and `ghc` existing to run its functionality.  Should I be able to invoke those from my test?

Reply:
If you've set the dependencies correctly, yes.